File: 88-MergeSortedArray/88-MergeSortedArray.cpp
// Last updated: 7/2/2025, 11:54:15 PM
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int s = m + n - 1;  // Correct total size
        m -= 1;
        n -= 1;

        // Merge from the end
        while (m >= 0 && n >= 0) {
            if (nums1[m] > nums2[n])
                nums1[s--] = nums1[m--];
            else
                nums1[s--] = nums2[n--];
        }

        // Copy any remaining nums2 elements
        while (n >= 0) nums1[s--] = nums2[n--];
    }
};

File: 88-MergeSortedArray/88-MergeSortedArray.py
# Last updated: 7/2/2025, 11:54:19 PM
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        total_len = n + m -1

        while m>0 and n>0:
            if nums1[m-1] > nums2[n-1]:
                nums1[total_len] = nums1[m-1]
                m-=1
            else:
                nums1[total_len] = nums2[n-1]
                n-=1
            total_len -= 1       
        
        while n>0:
            nums1[total_len] = nums2[n-1]
            n-=1
            total_len-=1
        
        
        